### PR TITLE
Documentation update (mostly to the developer section)

### PR DESCRIPTION
**Pull request by GernotMaier**
This PR review parts of the documentation and tries to improve it.

In detail:

- split developer guidelines into coding and developer guidelines (otherwise the page is too far long)
- review of developer section
- addition of pull request guidelines
- integration of recent code updates (e.g., geo_coordinates)
- improved getting_started .

For the review: it might be better to pull the branch locally, and run in the `doc` directory `make clean` and `make html`. Look at the files starting from `docs/build/html/index.html` and look only at the getting started, Developer Guidelines, Pull requests, and Coding guidelines pages. I think this is easier than trying to look at the github files changed page.

### CODE REVIEW

**Comment by GernotMaier** [comment]
Want to add: this is documentation only, so review it whenever it fits best in your time line.

File: docs/source/pull_requests.rst [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+Pull requests are one of our most important and powerful development tools.
+They help to obtain good quality, homogenous, and maintainable code.
+
+Both reviewing pull requests and addressing reviewers’ comments is hard.
+The following thoughts might help to obtain an efficient and pleasant reviewing process.
+
+Before issuing a pull request
+-----------------------------
+
```
Comment (VictorBarbosaMartins) [inline docs/source/pull_requests.rst:14]: I would add also here something about opening an issue beforehand and communication (see my previous comment about this) in coding_guidelines.rst

File: docs/source/coding_guidelines.rst [hunk 2]
```diff
@@ -0,0 +1,154 @@
+.. _CodingGuidelines:
+
+Coding Guidelines
+*****************
+
+This document describes the coding guidelines for the project. 
+
+In general, the simtools project follows the PEP8 style.
+It uses the same guidelines outlined in the `ctapipe style guide <https://cta-observatory.github.io/ctapipe/development/style-guide.html#>`_
```
Comment (VictorBarbosaMartins) [inline docs/source/coding_guidelines.rst:9]: both links are not working, could you please check it?

File: docs/source/coding_guidelines.rst [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+
+API documentation
+-----------------
+
+Docstrings following the Numpy style must be added to any public function, class or method.
+It is also recommended to add docstrings-like comments to private functions.
+
+In applications, the modules should contain docstrings with a general description, command line
+parameters, and examples.
```
Comment (VictorBarbosaMartins) [inline docs/source/coding_guidelines.rst:48]: I would mention also with errors that might be raised by the function

**Comment by VictorBarbosaMartins** [comment]
Ah, and following the guidelines themselves, I already committed several very simple grammar/typo things.

**Comment by GernotMaier** [comment]
@VictorBarbosaMartins  - thanks a lot for the suggestions!

I think I've followed almost all of them, see the last commit. Let me know if this is ok to merge.

(I am away now, so feel free to merge after you've approved).

File: docs/source/getting_started.rst [hunk 4]
```diff
@@ -3,150 +3,182 @@
...
+.. code-block:: console
 
-+++++++++++++++++++++++++++
-Set environmental variables
-+++++++++++++++++++++++++++
+    $ tar -czf corsika7.7_simtelarray.tar.gz
+    $ ./build_all prod5 qgs2 gsl
 
-Source the ``set_DB_environ.sh`` script (see `Model Database Access`_) to activate set the environmental variables for the DB access:
+Source the ``set_DB_environ.sh`` script (see `Model Database Access`_) to set the environmental variables required for database access:
```
Comment (orelgueta) [inline docs/source/getting_started.rst:157]: Do we plan to keep both the `set_DB_environ.sh` and the `.env` file? I thought we move completely to `.env`.

**Comment by VictorBarbosaMartins** [comment]
Hi @GernotMaier , everything good from my side, thanks for considering the suggestions. I will wait until you see Orel's comment about the .env file. Maybe we can still address that here if you like. Please, feel free to merge at any point.

**Pull request merged**